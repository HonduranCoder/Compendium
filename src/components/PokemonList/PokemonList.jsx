//Map through Pokemon array to display individual pokemon
//pokemonList.map(pokemon=>(<Pokemon {...props (pokemon.image)}/>))

import React from 'react';
import Pokemon from '../Pokemon/Pokemon.jsx';

export default function PokemonList({ pokemonList }) {
  //pass in props from the view, where state is going to live
  return (
    <div>
      <ul aria-label="pokemonList">
        {pokemonList.map(
          ({ pokemon, hp, attack, defense, shape, url_image }) => (
            <li key={pokemon}>
              <Pokemon
                image={url_image}
                pokemon={pokemon}
                hp={hp}
                attack={attack}
                defense={defense}
                shape={shape}
              />
            </li>
          )
        )}
      </ul>
    </div>
  );
}
