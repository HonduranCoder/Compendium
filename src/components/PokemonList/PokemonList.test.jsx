import { screen, render } from '@testing-library/react';
import PokemonList from './PokemonList.jsx';

it('test Pokemon List', () => {
  const data = [
    {
      image: 'http://assets.pokemon.com/assets/cms2/img/pokedex/full/617.png',
      pokemon: 'anorth',
      hp: '45',
      attack: '95',
      defense: '50',
      shape: 'armor',
    },
    {
      image: 'http://assets.pokemon.com/assets/cms2/img/pokedex/full/347.png',
      pokemon: 'accelgor',
      hp: '80',
      attack: '70',
      defense: '40',
      shape: 'arms',
    },
  ];
  render(<PokemonList pokemonList={data} />);
  const pokemonList = screen.getByLabelText('pokemonList');
  expect(pokemonList).toMatchSnapshot();
});
