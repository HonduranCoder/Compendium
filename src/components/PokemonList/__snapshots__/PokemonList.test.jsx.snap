// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Pokemon List 1`] = `
<ul
  aria-label="pokemonList"
>
  <li>
    <div
      aria-label="pokemon"
    >
      <img
        alt="poke"
      />
      <p>
        Name: 
        anorth
      </p>
      <p>
        HP: 
        45
      </p>
      <p>
        Attack: 
        95
      </p>
      <p>
        Defense: 
        50
      </p>
      <p>
        Shape: 
        armor
      </p>
    </div>
  </li>
  <li>
    <div
      aria-label="pokemon"
    >
      <img
        alt="poke"
      />
      <p>
        Name: 
        accelgor
      </p>
      <p>
        HP: 
        80
      </p>
      <p>
        Attack: 
        70
      </p>
      <p>
        Defense: 
        40
      </p>
      <p>
        Shape: 
        arms
      </p>
    </div>
  </li>
</ul>
`;
